### ISSUE TITLE
coordinate system in this project

### ISSUE THREAD

**Issue by murumura** [JulianKnodt/nerf_atlas#10]
I really appreciate for this project, which allows me to know many extensions of nerf without having to understand the coordinate system/render routine design by different authors... I can focus more on the model itself.

I found a little confused on derivation of dir/right/up vector part, which is opposite of the teaching I found on Scratch pixel (https://www.scratchapixel.com/lessons/mathematics-physics-for-computer-graphics/lookat-function) .
Could you please tell us about the coordinate system used or the purpose of such calculations?
```python
class OrthogonalCamera(Camera):
  def __init__(
    self,
    pos: torch.tensor,
    at: torch.tensor,
    up: torch.tensor,
    view_width:float=10.,
    view_height:float=None,
  ):
    super().__init__()
    if view_height is None: view_height = view_width
    self.pos = pos
    self.dir = dir = F.normalize(at - pos, dim=-1)
    self.right = r = view_width * F.normalize(torch.cross(dir, up), dim=-1)
    self.up = view_height * F.normalize(torch.cross(r, dir), dim=-1)
  def __len__(self): return self.pos.shape[0]
  def __getitem__(self, v):
    # FIXME to make this not use an odd part of the language? it should be fine
    cam = OrthogonalCamera.__new__()
    cam.pos = self.pos[v]
    cam.dir = self.dir[v]
    cam.right = self.up[v]
    cam.up = self.up[v]
    return cam
  def sample_positions(self, position_samples, size:int, with_noise=False):
    u, v = position_samples.split([1,1], dim=-1)
    u = 2 * ((u.squeeze(-1)/size) - 0.5)
    v = 2 * ((v.squeeze(-1)/size) - 0.5)
    r_o = self.pos[:, None, None, :] + \
      u[None, :, :, None] * self.right[:, None, None, :] +\
      v[None, :, :, None] * self.up[:, None, None, :]
    r_d = self.dir.expand_as(r_o)
    return torch.cat([r_o, r_d], dim=-1)
```
TIA.

**Comment by JulianKnodt** [comment]
For the orthogonal camera in specific, we define a look at transform, as specified by the position of the camera, the object being looked at, and an upwards direction.

(at - pos) returns the direction from the camera to the object, which is the forward pointing direction of the camera. Then, we cross the forward with the up direction, and using the right-hand rule this gives the right vector of the camera. Note that the forward, right and up directions may now not form an orthogonal basis, so it's necessary to re-adjust the up direction as the cross of the right and forward directions.

The positions are pretty naive, in that they are simple additions in the right/up plane where the camera is at.

Hopefully this helps, let me know if you have more questions

**Comment by murumura** [comment]
Thanks for the detailed reply !
According to the answer and my own understanding, this is to construct a camera-to-world matrix, where in camera coordinate the camera located at origin and facing toward +z axis (not -z as openGL coordinate? ), given
- pos : position of the camera in world coordinates
- at: position of the look-at object in world coordinates
- up: vector specifying the up direction in the world coordinate frame.

I draw a simplify diagram below to show my understanding:
![2021-11-24_12-27](https://user-images.githubusercontent.com/38805921/143175027-94f82d4c-51c1-4bc8-a178-4b59f7875776.png)

Is my understanding correct? Thanks a lot

**Comment by JulianKnodt** [comment]
Yep, something like this! I would suggest reading PBRT for more details on camera models!
https://pbr-book.org/3ed-2018/contents

**Comment by murumura** [comment]
thank again! I'm closing the issues!
